src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -253,6 +254,32 @@ private void ValidateDeprecated(JsonSchema schema,  JToken token, List<Error> er
             }
         }
 
+        private void ValidateDependentEnums(JsonSchema schema, JObject map, List<Error> errors)
+        {
+            foreach (var fieldName in schema.DependentEnums.Keys)
+            {
+                if (map.TryGetValue(fieldName, out var fieldValue) && fieldValue.Type == JTokenType.String)
```

Comment:
see `ValidateEnum`, the value does need to be a string, can just compare `JToken`